### ISSUE TITLE
~/Applications (non-admin)

### ISSUE THREAD

**Issue by devtobo** [andreyvit/create-dmg#54]
Is it possible to install my app in ~/Applications instead of /Applications?

**Comment by devtobo** [comment]
What happen when the ~/Applications directory doesn't exist?

**Comment by garethbrown** [comment]
I've got the same question. At the moment the application is installed as root. How do we make it install as the user?

**Comment by aonez** [comment]
Are you talking about the "Applications" alias? I don't think it can be dynamic, so going for the `/Applications` will be safer than `~/Applications`. But not sure here.

**Comment by apjanke** [comment]
Do you mean creating the DMG so that the "Applications" folder inside the DMG's display points to ~/Applications instead of /Applications? I don't think I've seen that before; do you have any examples of applications that do so?

I think this would be doable by providing an option like `--user-app-drop-link` to complement the current `--app-drop-link` option. Not sure how safe it would be. I *think* that ~/Applications is a standard directory that almost always exists, because it's created as part of the user's home directory skeleton when the user account is created. But I'm not sure.

**Comment by aonez** [comment]
@apjanke that folder does not come by default. Not 100% sure but my guess is that third party apps (like Chrome) create it.

I only can think of a script that:

1. Has applied the Applications icon, so it feels like a folder
2. Executing it (double click) creates the folder if needed and opens it in Finder
3. Executing it with an argument (probably dropped the application bundled in the DMG) does point 2 and copies the argument there

**Comment by apjanke** [comment]
> that folder does not come by default.

Yep, you right. I just created a new user `alice` on my macOS 10.14.6 box, and there is no `~/Applications` folder there in their new home folder.

I don't think we have the ability to execute arbitrary scripts etc for creating folders upon DMG opening or drag-and-drop, so this is smelling like a WontFix to me, at least for now.

If a user is sophisticated, they can always create `~/Applications` and open that in another Finder window and do the drag-and-drop from the DMG Finder window that way.

Objections or other thoughts?

**Comment by aonez** [comment]
I'll try the script thing to see how the quarantine affects to it, just for fun. Other than that seems like a no go.

**Comment by aonez** [comment]
Checked and the quarantine makes the script useless.
